### ISSUE TITLE
Unexpected inversion when using inverted registered outputs

### ISSUE THREAD

**Issue by and3rson** [daveho/GALasm#16]
Consider this example:

```pld
GAL22V10
PS2

CLK  A    B    NC   NC   NC   NC   NC   NC   NC   NC   GND
/CS  Q1   Q2   NC   NC   NC   NC   NC   NC   NC   NC   VCC

/Q1.R = A * B
/Q2.R = /Q1

DESCRIPTION
t
```

If I invert `Q1` and `Q2` in pin definitions as such:
```diff
< /CS  Q1   Q2   NC   NC   NC   NC   NC   NC   NC   NC   VCC
---
> /CS /Q1  /Q2   NC   NC   NC   NC   NC   NC   NC   NC   VCC
```
...I can observe the following difference in fuses:

```diff
132c132
< Pin 15 = Q2           S0 = 0   S1 = 0
---
> Pin 15 = /Q2          S0 = 1   S1 = 0
134c134
< 112  ---- ---- ---- ---- ---- ---- ---- ---- ---- --x- ----
---
> 112  ---- ---- ---- ---- ---- ---- ---- ---- ---- ---x ----
145c145
< Pin 14 = Q1           S0 = 0   S1 = 0
---
> Pin 14 = /Q1          S0 = 1   S1 = 0
```

I would expect only the S0 bit to change. However, the actual column is also getting changed (from true to inverted).

I checked the datasheet (http://web.mit.edu/6.115/www/document/gal22v10.pdf). On page 4 in the top section (S1=0) it's stated that the feedback originates from `/Q` while the pin itself is driven by `Q`, so pin inversion does not have any influence on the feedback since it's driven by a separate driver.
This means that `S0=0,S1=0` and `S0=1,S1=0` should only change the behavior of the physical output, not the matrix. In other words, inverting a pin in the definition section should only change the S0 flag.
Is this a bug or am I misunderstanding something?

**Comment by and3rson** [comment]
I believe it should be something like this:
```diff
diff --git a/src/galasm.c b/src/galasm.c
index 052ad70..7efa722 100644
--- a/src/galasm.c
+++ b/src/galasm.c
@@ -1598,11 +1598,14 @@ void SetAND(int row, int pinnum, int negation, int gal_type)
         case GAL22V10:
             column  =  PinToFuse22V10[pinnum - 1];

+            if ((pinnum >= 14 && pinnum <= 23)) {
                                     /* is it a registered OLMC pin?   */
                                     /* yes, then correct the negation */
-            if ((pinnum >= 14 && pinnum <= 23) && !Jedec.GALS1[23 - pinnum])
-            {
-                negation = negation ? 0 : 1;
+                if (!Jedec.GALS1[23 - pinnum]) {
+                    negation = 1;
+                } else {
+                    negation = 0;
+                }
             }

                        numofcol = MAX_FUSE_ADR22V10 + 1;
```

Feedbacks from registered OLMCs are always negated (they come from /Q) - see page 3 of datasheet:
> NOTE: In registered mode, the feedback is from the /Q output of the register, and not from the pin

**Comment by and3rson** [comment]
I think I figured it out.

GALasm implicitly changes S0 of OLMC if polarity of pin definition and pin equation changes.

e.g.
```
/CS /Q1 ; ...

/Q1.R = A * B

; Q0: S0 = 1, S1 = 0
```

```
/CS /Q1 ; ...

Q1.R = A * B

; Q0: S0 = 0, S1 = 1
```

**Comment by and3rson** [comment]
Another update. Consider the following equations:

```
/Q1.R = A * B
Q2.R = Q1
```

When compiled with `galasm` and then decompiled with `jed2eqn`, I get the following equations:

```
/Q1 := A * B
```Q2 := /Q1


I expected Q1 & Q2 to be equal, but they are not.

**Comment by and3rson** [comment]
I think the offending line is `galasm:1605` which shouldn't be there, since it inverts the pin (inside equation) if it is active-low. But being active-low should not affect any equations (as per the datasheet).

**Comment by pjdennis** [comment]
Fixed in commit c376d56
